bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1244 @@
+    private int volumeAbsValue = 0;
+    private @Nullable String responseCode = "";
+    private int volumeState = 0;
+    private int maxVolumeState = 0;
+    private @Nullable String inputState = "";
+    private @Nullable String soundProgramState = "";
+    private int sleepState = 0;
+    private @Nullable String artistState = "";
+    private @Nullable String trackState = "";
+    private @Nullable String albumState = "";
+    private @Nullable String repeatState = "";
+    private @Nullable String shuffleState = "";
+    private int playTimeState = 0;
+    private int totalTimeState = 0;
+    private @Nullable String zone = "main";
+    private String channelWithoutGroup = "";
+    private @Nullable String thingLabel = "";
+    private @Nullable String mclinkSetupServer = "";
+    private @Nullable String mclinkSetupZone = "";
+    private String url = "";
+    private String json = "";
+    private String action = "";
+    private int zoneNum = 0;
+    private @Nullable String groupId = "";
+    private @Nullable String host;
+    public @Nullable String deviceId = "";
+
+    private YamahaMusiccastStateDescriptionProvider stateDescriptionProvider;
+
+    public YamahaMusiccastHandler(Thing thing, YamahaMusiccastStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String localValueToCheck = "";
+        String localRole = "";
+        boolean localSyncVolume;
+        String localDefaultAfterMCLink = "";
+        String localRoleSelectedThing = "";
+        if (command != RefreshType.REFRESH) {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            channelWithoutGroup = channelUID.getIdWithoutGroup();
+            zone = channelUID.getGroupId();
+            DistributionInfo distributioninfo = new DistributionInfo();
+            Response response = new Response();
+            switch (channelWithoutGroup) {
+                case CHANNEL_POWER:
+                    if (command == OnOffType.ON) {
+                        httpResponse = setPower("on", zone, this.host);
+                        response = gson.fromJson(httpResponse, Response.class);
+                        if (response != null) {
+                            localValueToCheck = response.getResponseCode();
+                            if (!"0".equals(localValueToCheck)) {
+                                updateState(channelUID, OnOffType.OFF);
+                            }
+                        }
+                        // check on scheduler task for UDP events
+                        ScheduledFuture<?> localGeneralHousekeepingTask = generalHousekeepingTask;
+                        if (localGeneralHousekeepingTask == null) {
+                            logger.info("YXC - No scheduler task found!");
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug or warn. See https://next.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.

Reply:
Only 2 occurences, changed it to debug.

Reply:
Updated, push already done.